src/Analysis/Ast/Impl/Analyzer/Definitions/IPythonAnalyzer.cs

```diff
@@ -56,10 +56,12 @@ public interface IPythonAnalyzer {
         /// </summary>
         IReadOnlyList<DiagnosticsEntry> LintModule(IPythonModule module);
 
+
         /// <summary>
-        /// Removes all the modules from the analysis, except Typeshed and builtin
+        /// Removes all the modules from the analysis and restarts it.
         /// </summary>
-        void ResetAnalyzer();
+        /// <param name="full">True if everything should be dropped, including closed files and stubs.</param>
+        Task ResetAnalyzer(bool full);
```

Comment:
I had to modify this interface method. Should I duplicate it with another signature to ensure the old one is retained, or is this alright?

Reply:
If we need to reload builtins on `NotifyPackagesChanged`, we probably should do the same on `DidChangeConfiguration`, so there is no reason for the `full == false` case.

Reply:
In its current form, HandleConfigurationChanges only returns true if the diagnostics settings change. I can unconditionally do a full reload in that case, but we weren't before so I didn't want to change it. I just know that we don't want to do a reload on _any_ config change, as only some impact the analysis.

Reply:
Actually, I'm surprised that the diagnostics settings cause any reloads at all. My impression was that changing them only affected the layer that exposes them to the editor. We could probably notify that of a change to have it re-filter and re-send things rather than a full reanalysis (nothing changed except the filters).

Reply:
Yeah diag changes should not be reload

Reply:
My thoughts were correct, the non-full branch is basically useless and doesn't need to be done at all, so I can clean this up and eliminate the boolean arg.